lib/ansible/modules/yum_repository.py

```diff
@@ -657,6 +676,69 @@ def main():
     if module.params['params']:
         module.fail_json(msg="The params option to yum_repository was removed in Ansible 2.5 since it circumvents Ansible's option handling")
 
+    if module.params['name'] is None and module.params['copr'] is None:
+        module.fail_json(msg="Either the 'name' or the 'copr' parameter needs to be set")
+
+    if module.params['copr']:
+        copr_api = module.params['copr_api']
+        try:
+            copr_user, copr_project = module.params['copr'].split('/')
```

Comment:
It might be nice to be explicit here just to be safe, even though copr currently prevents `/` in names.

```suggestion
            copr_user, copr_project = module.params['copr'].split('/', 1)
```